crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -394,6 +405,15 @@ pub(crate) fn dynamic_resolution_paths(db: &dyn Db) -> Vec<SearchPath> {
         }
     }
 
+    // Append the real stdlib as the very last option in search.
+    // Normally this means it will always be shadowed by typeshed.
+    //
+    // FIXME(Gankra): ideally this should be completely disabled unless we're in
+    // `ModuleResolveMode::NoStubsAllowed`.
+    if let Some(real_stdlib_path) = real_stdlib_path {
+        dynamic_paths.push(real_stdlib_path.clone());
+    }
```

Comment:
hmm, I'm not sure it's correct for this to come after e.g. the `site-packages` search path. I think this would mean that if e.g. the user had the moribund [`asyncio` package from PyPI](https://pypi.org/project/asyncio/) installed into their venv, they had `import asyncio` in a file, and they did go-to-definition on the `asyncio` symbol, we'd jump to the module in the venv's site-packages directory. But at runtime, the module in the venv's site-packages directory will be totally ignored: the stdlib takes precedence in module resolution over third-party packages, so `import asyncio` will resolve to the stdlib `asyncio` module.

Reply:
oh sorry, I posted this comment before I saw your comment above

Reply:
No this is still a great insight, thanks!